designs/2023-rule-performance-dashboard/README.md

```diff
@@ -0,0 +1,276 @@
+- Repo: eslint/eslint
+- Start Date: 2023-03-10
+- RFC PR: <https://github.com/eslint/rfcs/pull/108>
+- Authors: Mara Nikola Kiefer (@mnkiefer)
+
+# Performance Statistics
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+This document describes a new ESLint option `stats`, which enables the user to obtain a series of runtime *statistics* on top of their final lint results.
+
+These statistics include more granular information, such as the [parse](#timing-parse)-, [fix](#timing-fix)-, and [lint-times](#timing-lint) and the number of [fix passes](#fix-passes), [directives and violations](#directives-and-violations) encountered.
+
+| Stats |  | Description |
+|--------|-------|---|
+| **timing** | total  | The overall lint time that is spent on a file. This includes `parse-`, `fix-` and all `rules-` (total) times. |
+|        | parse | The time that is spent when parsing a file, that is when [`parse()`](#timing-parse) or `parseForEslint()` is called. |
+|        | rules | This is similar to the rule performance TIMING, except that it is more granular, that is, persisted as *per file per rule*.  |
+| **passTiming** | | An array of *timing* objects for each of the [fix passes](#fix-passes). In total, these amount to the (total) *timing* object for the respective file. |
+| **fixPasses** | | The number of [fixes](https://eslint.org/docs/latest/use/command-line-interface#fix-problems) successfully applied to a file. |
```

Comment:
I'm confused by this name `fixPasses` and how it corresponds to `the number of fixes successfully applied to a file`?

Reply:
This property was also by @nzakas. It goes into [this loop](https://github.com/mnkiefer/eslint/pull/1/files#diff-a4ade4bc7a7214733687d84fbb54b625e464d13be7181caf54f564e5985db980R2057) and counts the number of  passing during autofixing.

Reply:
> And I imagine the motivation for including this metric is since the statistics are totaled across passes, knowing the number of passes would allow you to compute an average per pass.

Yes, that's correct. I think we can clean up the language a bit to make it clearer. Maybe something like this?

```suggestion
| **fixPasses** | | The number of times ESLint has applied at least one fix after linting. |
```